analysis/study_definition.py

```diff
@@ -1,13 +1,344 @@
+######################################
 
+# This script provides the formal specification of the study data that will be extracted from
+# the OpenSAFELY database.
 
+######################################
+
+
+# IMPORT STATEMENTS ----
+
+# Import code building blocks from cohort extractor package
+from cohortextractor import (
+    StudyDefinition,
+    patients,
+    Measure,
+    codelist,
+)
+
+# Import codelists from codelist.py (which pulls them from the codelist folder)
+from codelists import *
+
+
+## Define study population and variables
 study = StudyDefinition(
-    default_expectations={
-        "date": {"earliest": "1900-01-01", "latest": "today"},
-        "rate": "uniform",
-        "incidence": 0.5,
+
+  # Set index date
+  index_date = "2018-01-01",
+  
+  # Configure the expectations framework
+  default_expectations={
+    "date": {"earliest": "2018-01-01", "latest": "2022-03-31"},
+    "rate": "uniform",
+    "incidence": 0.1,
+  },
+ 
+  # Define the study population
+  # TODO: for TPP studies, do we need to specify TPP registered?
+  population = patients.satisfying(
+      """
+      NOT has_died
+      AND
+      registered
+      AND 
+      (sex = "M" OR sex = "F")
+      AND
+      (age >=18 AND age < 110)
+      """,
+    
+      has_died = patients.died_from_any_cause(
+        on_or_before = "index_date",
+        returning = "binary_flag",
+      ),
+    
+      registered = patients.satisfying(
+       "registered_at_start",
+       registered_at_start = patients.registered_as_of("index_date"),
+      ), 
+    ),
+  
+  ## Variables
+
+  ### Sex
+  sex = patients.sex(
+    return_expectations = {
+      "rate": "universal",
+      "category": {"ratios": {"M": 0.49, "F": 0.51}},
+    },
+  ),
+  
+  ### Age
+  age = patients.age_as_of(
+    "index_date",
+    return_expectations = {
+      "rate": "universal",
+      "int": {"distribution": "population_ages"},
+      "incidence" : 0.001
+    },
+  ),
+
+  ### Age categories
+  age_cat = patients.categorised_as(
+    {"0": "DEFAULT",
+      "1": """age >= 18 AND age < 30""",
+      "2": """age >= 30 AND age < 40""",
+      "3": """age >= 40 AND age < 50""",
+      "4": """age >= 50 AND age < 60""",
+      "5": """age >= 60 AND age < 70""",
+      "6": """age >= 70 AND age < 80""",
+      "7": """age >= 80 AND age < 90""",
+      "8": """age >= 90""",
+    },
+
+    return_expectations = {
+      "rate": "universal",
+      "category": {
+        "ratios": {
+          "0": 0.01,
+          "1": 0.16,
+          "2": 0.18,
+          "3": 0.16,
+          "4": 0.18,
+          "5": 0.14,          
+          "6": 0.11,
+          "7": 0.05,
+          "8": 0.01,
+        }},
+    },
+  ),
+  
+  ### Index of multiple deprivation
+  imd = patients.categorised_as(
+    {"0": "DEFAULT",
+      "1": """index_of_multiple_deprivation >=1 AND index_of_multiple_deprivation < 32844*1/5""",
```

Comment:
Should be >=0 as described [here](https://docs.opensafely.org/study-def-tricks/#grouping-imd-by-quintile)